### PR TITLE
Refactor/fix clippy lints

### PR DESCRIPTION
**Pull request by shogo-nakano-desu**
As title says.

### CODE REVIEW

File: crates/flycheck/src/lib.rs [hunk 1]
```diff
@@ -533,7 +535,7 @@ impl CargoActor {
 }
 
 enum CargoMessage {
-    CompilerArtifact(cargo_metadata::Artifact),
+    CompilerArtifact(Box<cargo_metadata::Artifact>),
```
Comment (lnicola) [inline crates/flycheck/src/lib.rs:538]: Last time this came up (can't find the PR, though), it was pointed out that this is the most common variant, so boxing the contents will cause almost every instance to incur an allocation.

I don't know if this overhead will be larger than the speed-up obtained from the faster moves. Either way, I suspect it doesn't matter too much compared to the whole `cargo check` invocation.

**Review by lnicola — APPROVED** [review]

File: crates/flycheck/src/lib.rs (hunk 1 above)
Comment (shogo-nakano-desu) [reply crates/flycheck/src/lib.rs:538]: Ah, I see. I will keep it in my mind. Thank you for your review.

Comment (shogo-nakano-desu) [reply crates/flycheck/src/lib.rs]: @lnicola I have removed Box in
[dd84306](https://github.com/rust-lang/rust-analyzer/pull/15615/commits/dd843060f9ebff3862dc48c033c3ddbabb3b6a84) . Thanks

**Comment by Veykril** [comment]
Thanks!
@bors r+
On that note, I believe the lints section in Cargo.toml has been set for stabilization now, https://github.com/rust-lang/cargo/issues/12115 so we can soon enable clippy in general.

**Pull request merged**